packages/core/lib/queues/AbstractPublisherManager.ts

```diff
@@ -1,25 +1,18 @@
-import type {
-  AsyncPublisher,
-  CommonEventDefinition,
-  QueuePublisherOptions,
-} from '@message-queue-toolkit/core'
 import type { TypeOf, z } from 'zod'
 
 import type { EventRegistry } from '../events/EventRegistry'
 import type { BaseEventType } from '../events/baseEventSchemas'
+import type { CommonEventDefinition } from '../events/eventTypes'
 import type { MetadataFiller } from '../messages/MetadataFiller'
-import type { SyncPublisher } from '../types/MessageQueueTypes'
-import type { CommonCreationConfigType } from '../types/queueOptionsTypes'
+import type { AsyncPublisher, SyncPublisher } from '../types/MessageQueueTypes'
+import type { CommonCreationConfigType, QueuePublisherOptions } from '../types/queueOptionsTypes'
 
-export type MessagePublishType<T extends CommonEventDefinition> = Pick<
-  z.infer<T['schema']>,
-  'type' | 'payload'
-> & { id?: string }
+export type MessagePublishType<T extends CommonEventDefinition> = z.infer<T['schema']>
```

Comment:
Unfortunately, previous Pick-based implementation was breaking payload type inference during publishing. TS started accepting any combination of valid message types and payloads, and not enforcing that they belong to the same message